Minor enhancements/improvements for offline autoinstall.

SebastianObi opened the pull request:
- Also use "--nocheck" parameter with autoinstall.
- Added "--ignoredownload" parameter which try download latest firmware but fallback to local firmware.
- Added "--download" parameter for downloading all available firmwares for later offline usage.

SebastianObi committed:
Minor enhancements/improvements for offline autoinstall.

- Also use "--nocheck" parameter with autoinstall.
- Added "--ignoredownload" parameter which try download latest firmware but fallback to local firmware.
- Added "--download" parameter for downloading all available firmwares for later offline usage.

SebastianObi commented:
Hi Mark,

if you think my changes make sense then feel free to adopt them. You can also rename the parameters if you think of a better name.

markqvist commented:
Thanks for this, that is some really useful additions. I will merge these features as soon as I have time to check it all out and verify that it works for all the different board versions and such.

SebastianObi committed:
Update mechanism improved: Use of ZIP file for offline update.

SebastianObi committed:
Small improvements for download and installation with own firmware server

Added new parmeter "--url": Defines an alternative firmware download URL

Improved download messages

markqvist commented:
Rejected due to conflicting interest with the overall purpose of the Reticulum project.

In the interest of general transparency, see the following discussion thread: https://github.com/markqvist/Reticulum/discussions/127#discussioncomment-3959879
